dbms/src/Storages/StorageFactory.cpp

```diff
@@ -255,6 +258,44 @@ StoragePtr StorageFactory::get(
 			materialized_columns, alias_columns, column_defaults,
 			attach, context.getSettings().max_compress_block_size);
 	}
+	else if (name == "File")
+	{
+		ASTs & args_func = typeid_cast<ASTFunction &>(*typeid_cast<ASTCreateQuery &>(*query).storage).children;
+
+		constexpr auto error_msg = "Storage File requires exactly 1 parameter - name of using format.";
+
+		/// TODO: Maybe some my misunderstanding of ASTs
+
+		if (args_func.size() != 1)
+			throw Exception(error_msg, ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+
+		ASTs & args = typeid_cast<ASTExpressionList &>(*args_func.at(0)).children;
+
+		if (args.empty() || args.size() > 2)
+			throw Exception(error_msg, ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+
+		args[0] = evaluateConstantExpressionOrIdentidierAsLiteral(args[0], local_context);
+		String format_name = static_cast<const ASTLiteral &>(*args[0]).value.safeGet<String>();
+
+		int source_fd = -1;
+		String source_path;
+		if (args.size() >= 2)
+		{
+			args[1] = evaluateConstantExpressionOrIdentidierAsLiteral(args[1], local_context);
+			String table_source = static_cast<const ASTLiteral &>(*args[1]).value.safeGet<String>();
+
+			if (table_source == "stdin")
+				source_fd = STDIN_FILENO;
+			else
+				source_path = std::move(table_source);
```

Comment:
And also nice to have possibility to specify file descriptors as numbers.
Example:

```
clickhouse-local --query="CREATE TABLE ... ENGINE = File(TabSeparated, 3); SELECT * FROM table;" 3<file
```

So you could read from many file descriptors.
Quite obscure feature, but usable in shell scripts.